### DISCUSSION TITLE
Adding a plugin

### DISCUSSION THREAD

**Discussion by SebastianSlash** [Doraku/DefaultDocumentation#105] [Q&A] [answered]
I have tried to add the plugin sample to my project because I wanted to add a yaml header section to all markdown files. Sadly I have failed, somehow I'm not able to include it.
Also, I currently candy share any code or error messages because the project is at work and I don't have a work GitHub account.

Anyways, this is what I've done:

- [Create a .NET windows forms hello world project](https://docs.microsoft.com/en-us/visualstudio/ide/create-csharp-winform-visual-studio?view=vs-2022)
- configure it to create a XML documentation file in debug
- install the DefaultDocumentation NuGet package for the project
- paste the example plugin project folder inside the hello world project folder
- right click on the hello world solution and add the example plugin project
- right click on the hello world project and add a reference to the example plugin project
- add <DefaultDocumentationSections>New|Header|Default</DefaultDocumentationSections> to the .target file in the example plugin project

That should be it. Then I get an error that the section "New" is not known/found

I could not find how to properly do this in the documentation here on GitHub.

Has anyone successfully added a custom section? The background is that i want to publish this documentation on GitHub Pages with Jekyll.

**Comment by Doraku** [comment, marked as answer]
Sorry I completely forgot to update the plugin sample after releasing the new version. If you use a plugin from the same solution as your project, you should first check that the plugin project and your own project reference the same version of DefaultDocumentation package (`DefaultDocumentation` for the project, `DefaultDocumentation.Api` for the plugin, the sample of this repo referenced the beta package of the api...).
Then you should add a reference to your compiled plugin in a `<DefaultDocumentationPlugins>` property of your csproj like so (or use the absolute path if you are not sure):
```xml
...
<PropertyGroup>
<DefaultDocumentationPlugins>$(SolutionDir)DefaultDocumentation.PluginExample/lib/netstandard2.0/DefaultDocumentation.PluginExample.dll</DefaultDocumentationPlugins>
</PropertyGroup>
...
```

I have updated the readme here with those info. Tell me if you still have issues.

**Reply by SebastianSlash** [reply]
That's it, thank you! Now I was able to get it to work. I'll see if I can publish what I did for the plugin here on GitHub so when someone needs to add a YAML header for Jekyll there is an example.
